types/tx_msg.go

```diff
@@ -11,27 +13,34 @@ type (
 	Msg interface {
 		proto.Message
 
-		// Return the message type.
-		// Must be alphanumeric or empty.
-		Route() string
-
-		// Returns a human-readable string for the message, intended for utilization
-		// within tags
-		Type() string
-
 		// ValidateBasic does a simple validation check that
 		// doesn't require access to any other information.
 		ValidateBasic() error
 
-		// Get the canonical byte representation of the Msg.
-		GetSignBytes() []byte
-
 		// Signers returns the addrs of signers that must sign.
 		// CONTRACT: All signatures must be present to be valid.
 		// CONTRACT: Returns addrs in some deterministic order.
 		GetSigners() []AccAddress
 	}
 
+	// LegacyMsg defines the old interface a message must fulfill, containing
+	// Amino signing method and legacy router info.
+	// Deprecated: Please use `Msg` instead.
+	LegacyMsg interface {
```

Comment:
I was wondering, could this be removed and go inside auth/legacy? Since from my understanding this is needed only for amino signing.

Reply:
> Since from my understanding this is needed only for amino signing.

Yes, the `.GetSignBytes` is used for amino. The `.Route` method should not be used anymore, as all Msgs are routed by the msg_service_router. The `.Type` is still used though, in events and logs.

I still moved it into `auth/legacy/legacytx`, lmk if that sounds right to you.